include/eve/module/core/regular/logical_and.hpp

```diff
@@ -83,32 +74,24 @@ namespace eve
 
   namespace detail
   {
-    template<typename T, typename U, callable_options O>
-    EVE_FORCEINLINE constexpr auto
-    logical_and_(EVE_REQUIRES(cpu_), O const &, T a, U b) noexcept
+    template<callable_options O, typename T, typename U>
+    EVE_FORCEINLINE constexpr common_logical_t<T, U> logical_and_(EVE_REQUIRES(cpu_), O const&, T a, U b) noexcept
+      requires (relaxed_logical_scalar_value<T> && relaxed_logical_scalar_value<U>) || (std::same_as<T, U>)
     {
-      using r_t = as_logical_t<decltype(a && b)>;
-      if constexpr( scalar_value<T> && scalar_value<U> ) return r_t(a && b);
-      else return a && b;
+      if  constexpr (relaxed_logical_scalar_value<T>) return a && b;
+      else                              return bit_cast(a.bits() & b.bits(), as<T>{});
```

Comment:
I don't understand this `else`. You are in `cpu` right? 

Is this like what - `logical_value` && `is_wide_logical`? Can you be more explicit about that, this is very confusing.

At least a comment.